declaration/models.py

```diff
@@ -47,10 +47,25 @@ class Meta:
 
     def __str__(self):
         return f'User: {self.user} - tracking code: {self.tracking_code}'
+    
+    def save(self, *args, **kwargs):
+        is_new_declaration = False
+        old_declaration = None
+        
+        if self.pk is None:
+            is_new_declaration = True
+        else:
+            old_declaration = PackageDeclaration.objects.get(pk=self.pk)
+
+        super(PackageDeclaration, self).save(*args, **kwargs)
+
+        if is_new_declaration or old_declaration.status != self.status:
```

Comment:
Also, write test units for this case, make sure statushistory is created and is correct